user-service/src/main/java/org/arpha/exception/handler/UserExceptionHandler.java

```diff
@@ -0,0 +1,72 @@
+package org.arpha.exception.handler;
+
+import jakarta.validation.ConstraintViolation;
+import lombok.extern.slf4j.Slf4j;
+import org.arpha.exception.EmailAlreadyTakenException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ProblemDetail;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.util.List;
+import java.util.Map;
+
+@RestControllerAdvice
+@Slf4j
+public class UserExceptionHandler extends ResponseEntityExceptionHandler {
+
+    private static final String MESSAGE = "message";
+
+    @ExceptionHandler(EmailAlreadyTakenException.class)
+    public ProblemDetail handleEmailAlreadyTakenException(EmailAlreadyTakenException e) {
+        log.error(MESSAGE, e);
+        return ProblemDetail.forStatusAndDetail(HttpStatus.BAD_REQUEST, e.getMessage());
+    }
+
+    @ExceptionHandler({AccessDeniedException.class, AuthenticationException.class})
+    public ProblemDetail handleAuthError(RuntimeException e) {
+        log.error(MESSAGE, e);
+        return ProblemDetail.forStatusAndDetail(HttpStatus.FORBIDDEN, e.getMessage());
+    }
+
+    @ExceptionHandler(RuntimeException.class)
+    public ProblemDetail handleUnexpectedExceptions(RuntimeException e) {
+        log.error(MESSAGE, e);
+        return ProblemDetail.forStatusAndDetail(HttpStatus.INTERNAL_SERVER_ERROR, e.getMessage());
```

Comment:
Consider not returning the actual error message to the user, since it can reveal some internal application information like db table names etc.